actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -904,9 +904,8 @@ def form_tag_with_body(html_options, content)
           output.safe_concat("</form>")
         end
 
-        # see http://www.w3.org/TR/html4/types.html#type-name
         def sanitize_to_id(name)
-          name.to_s.delete(']').tr('^-a-zA-Z0-9:.', "_")
+          name.to_s.delete(']').gsub(/[^-:.\p{Word}]/, "_")
```

Comment:
You can use `/[^-:.[:word:]]/` to match the current implementation of `ActionView::Helpers::Tags::Base#sanitized_value`. Is there a way to do this with `#tr`?